File: schemas/digital-asset-management/drizzle/workspaces.ts
// workspaces: Top-level organizational containers for digital assets.
// See README.md for full design rationale.

import { pgTable, uuid, text, bigint, timestamp } from "drizzle-orm/pg-core";
import { users } from "../../auth-rbac/drizzle/users";

export const workspaces = pgTable("workspaces", {
  id: uuid("id").primaryKey().defaultRandom(),
  name: text("name").notNull(),
  slug: text("slug").unique().notNull(),
  description: text("description"),
  logoUrl: text("logo_url"),
  storageLimitBytes: bigint("storage_limit_bytes", { mode: "number" }),
  storageUsedBytes: bigint("storage_used_bytes", { mode: "number" }).notNull().default(0),
  createdBy: uuid("created_by").notNull().references(() => users.id, { onDelete: "restrict" }),
  createdAt: timestamp("created_at", { withTimezone: true }).notNull().defaultNow(),
  updatedAt: timestamp("updated_at", { withTimezone: true }).notNull().defaultNow(),
});

File: schemas/digital-asset-management/spacetimedb/workspaces.rs
// workspaces: Top-level organizational unit for asset management.
// See README.md for full design rationale.

use spacetimedb::Timestamp;

#[spacetimedb::table(name = workspaces, public)]
pub struct Workspace {
    #[primary_key]
    pub id: String, // UUID

    pub name: String,

    #[unique]
    pub slug: String,

    pub description: Option<String>,
    pub logo_url: Option<String>,

    pub storage_limit_bytes: Option<i64>,
    pub storage_used_bytes: i64, // default 0

    pub created_by: String, // FK → users.id (restrict delete)

    pub created_at: Timestamp,
    pub updated_at: Timestamp,
}
